File: vm_Assignments/vmc/L2.3/RocketShipLong.c
#include<stdio.h>
void cone_part(int n)
{
    if(n%2 !=0)
    {
        int center = (n+1)/2;
        for(int i=1; i<=center; i++)
            printf(" ");
            printf("^\n");
            for(int j=1; j<center; j++)
            {
                for(int i=1; i<=center-j; i++)
                {
                    printf(" ");
                }
                for(int i=1; i<=j; i++)
                {
                    printf("%c", '/');
                }
                printf("%c", '|');
                for(int i=1; i<=j; i++)
                {
                    printf("%c", '\\');
                }
                printf("\n");
                }
            }
}
void body(int n)
{

    for(int i=0; i<4; i++)
    {
        printf("+");
        for(int j=0;j<=n-1 ;j++)
            printf("*");
        printf("+\n");
    }
    printf("+");
    for(int j=0;j<=n-1 ;j++)
        printf("-");
        printf("+\n");

    for(int i=0; i<4; i++)
    {
        printf("+");
        for(int j=0;j<=n-1 ;j++)
            printf("*");
        printf("+\n");
    }
        printf("+");
    for(int j=0;j<=n-1 ;j++)
        printf("-");
        printf("+\n");
}

void main()
{
    cone_part(7);
    body(7);
    cone_part(7);
}

File: vm_Assignments/vm/Beginning/L2.3/RocketShipLong.py
# Divide and Conqure

def print_char(c, char):
    for i in range(c):
        print(char, end="")

# w must be odd
def make_head(w):
    center = int((w+1)/2)
    print(" "*(center-1) + "^")
    for i in range(1,center-1):
        print_char(center-i-1, " ")
        print_char(i, "/")
        print("|", end="")
        print_char(i, "\\")
        print()
        
def make_body(w,l):
    for i in range(l):
        print("+" + "*"*(w-2) + "+")
    make_separator(w)

def make_separator(w):
    print("+" + "-"*(w-2) + "+")
        
# w: rocket width
# c: rocket body count
# l: rocket body length
def make_rocket(w,c,l): 
    make_head(w)
    make_separator(w)

    for i in range(c):
        make_body(w,l)

    make_head(w)

make_rocket(11,2,5)